drivers/spi/spi_ll_stm32.c

```diff
@@ -178,22 +244,48 @@ static void spi_stm32_complete(struct spi_stm32_data *data, SPI_TypeDef *spi,
 			       int status)
 {
 #ifdef CONFIG_SPI_STM32_INTERRUPT
+#ifdef CONFIG_SOC_SERIES_STM32MP1X
+	LL_SPI_DisableIT_TXP(spi);
+	LL_SPI_DisableIT_RXP(spi);
+	LL_SPI_DisableIT_DXP(spi);
+	LL_SPI_DisableIT_EOT(spi);
+	LL_SPI_DisableIT_TXTF(spi);
+	LL_SPI_DisableIT_UDR(spi);
+	LL_SPI_DisableIT_OVR(spi);
+	LL_SPI_DisableIT_CRCERR(spi);
+	LL_SPI_DisableIT_FRE(spi);
+	LL_SPI_DisableIT_MODF(spi);
+	LL_SPI_DisableIT_TSER(spi);
+#else
 	LL_SPI_DisableIT_TXE(spi);
 	LL_SPI_DisableIT_RXNE(spi);
 	LL_SPI_DisableIT_ERR(spi);
-#endif
+#endif /* CONFIG_SOC_SERIES_STM32MP1X */
+#endif /* CONFIG_SPI_STM32_INTERRUPT */
 
 	spi_context_cs_control(&data->ctx, false);
 
 #if defined(CONFIG_SPI_STM32_HAS_FIFO)
 	/* Flush RX buffer */
+#ifdef CONFIG_SOC_SERIES_STM32MP1X
+	while (LL_SPI_IsActiveFlag_RXP(spi)) {
+		(void)LL_SPI_ReceiveData8(spi);
+	}
+#else
 	while (LL_SPI_IsActiveFlag_RXNE(spi)) {
 		(void) LL_SPI_ReceiveData8(spi);
 	}
-#endif
+#endif	/* CONFIG_SOC_SERIES_STM32MP1X */
+#endif	/* CONFIG_SPI_STM32_HAS_FIFO */
 
 	if (LL_SPI_GetMode(spi) == LL_SPI_MODE_MASTER) {
+#ifdef CONFIG_SOC_SERIES_STM32MP1X
+		while (LL_SPI_IsEnabled(spi) &&
+		       !LL_SPI_IsActiveFlag_MODF(spi) &&
```

Comment:
Not sure about these conditione, if we check what is done in the while block

Reply:
This is just an adaptation to the BSY flag which isn't existing on STM32MP1 SPI peripheral registers.
Definition taken from _stm32l4xx_ll_spi.h_ : 
```
/**
  * @brief  Get busy flag
  * @note   The BSY flag is cleared under any one of the following conditions:
  * -When the SPI is correctly disabled
  * -When a fault is detected in Master mode (MODF bit set to 1)
  * -In Master mode, when it finishes a data transmission and no new data is ready to be
  * sent
  * -In Slave mode, when the BSY flag is set to '0' for at least one SPI clock cycle between
  * each data transfer.
  * @rmtoll SR           BSY           LL_SPI_IsActiveFlag_BSY
  * @param  SPIx SPI Instance
  * @retval State of bit (1 or 0).
  */
```
Regarding the while loop content, this is something that is introduced here https://github.com/zephyrproject-rtos/zephyr/pull/17804